scripts/west_commands/runners/mdb.py

```diff
@@ -0,0 +1,127 @@
+# Copyright (c) 2020 Synopsys.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+'''Runner for Synopsys Metaware Debugger(mdb).'''
+
+
+from os import path
+from runners.core import ZephyrBinaryRunner, RunnerCaps
+
+class MdbBinaryRunner(ZephyrBinaryRunner):
+    '''Runner front-end for mdb.'''
+
+    def __init__(self, cfg, cores=1, jtag='digilent', nsim_args='', gui=True,
+                dig_device=''):
+        super(MdbBinaryRunner, self).__init__(cfg)
+        self.jtag = jtag
+        self.cores = cores
+        if nsim_args != '':
+            self.nsim_args = path.join(cfg.board_dir, 'support', nsim_args)
+        else:
+            self.nsim_args = ''
+        self.gui = gui
+        self.elf_name = cfg.elf_file
+        self.dig_device = dig_device
+        self.commander = 'mdb'
+
+    @classmethod
+    def name(cls):
+        return 'mdb'
+
+    @classmethod
+    def capabilities(cls):
+        return RunnerCaps(commands={'flash', 'debug'})
+
+    @classmethod
+    def do_add_parser(cls, parser):
+        parser.add_argument('--jtag', default='digilent',
+                            help='choose the jtag interface')
+        parser.add_argument('--cores', default=1,
+                            help='choose the cores to debug')
+        parser.add_argument('--gui', default=True,
+                            help='enable gui debug')
+        parser.add_argument('--nsim_args', default='',
+                            help='if given, arguments for nsim simulator')
+        parser.add_argument('--dig-device', default='',
+                            help='''choose the digilent device when
+                             --jtag=digilent''')
+
+    @classmethod
+    def create(cls, cfg, args):
+        return MdbBinaryRunner(
+            cfg,
+            cores=args.cores,
+            jtag=args.jtag,
+            nsim_args=args.nsim_args,
+            gui=args.gui,
+            dig_device=args.dig_device
+            )
+
+    def do_run(self, command, **kwargs):
+        self.require(self.commander)
+
+        mdb_basic_options = ['-nooptions', '-nogoifmain',
+                            '-toggle=include_local_symbols=1']
+
+        # nsim
+        if self.nsim_args != '':
+            mdb_target = ['-nsim', '@'+self.nsim_args]
+        # hardware target
+        else:
+            if self.jtag == 'digilent':
```

Comment:
1. Even though one still may use non-Digilent JTAG (i.e. Ashling Opella-XD) but I guess it's quite a rare case given Opella's price tag and setup complexities. So I'd just use 2 options `-sim` (we don't have any other simulator other than nSIM to use with MDB so knowledgeable people will understand what we mean, those with no deep knowledge of proprietary ARC tools would better go w/o mention of nSIM to not add confusion as here in Zephyr we use nSIM as a separate runner) & `-hw` (which is an opposite to simulation).
2. Similarly to what I've done to OpenOCD runner I'd like to see it here as well: possibility to specify Digilent instance we're going to use. Otherwise we won't be able to use a host with multiple targets attached. For the record it's done with `-prop=dig_device=XXX` (it's important to note though that this is not the same Digilent serial number we use in OpenOCD runner but so-called "User Name", see below):
```
$ djtgcfg enum
 
Found 2 device(s)
 
Device: IoTDK
    Product Name:   DesignWare ARC SDP
    User Name:      IoTDK
    Serial Number:  251642517558
 
Device: HSDK
    Product Name:   DesignWare ARC SDP
    User Name:      HSDK
    Serial Number:  25164200001f
```